### DISCUSSION TITLE
Quick tip: One keymap can save you from using multiple plugins (tabout, vim-matchup,  nvim-treesitter-textobjects)

### DISCUSSION THREAD

**Discussion by charliie-dev** [ayamir/nvimdots#361] [Ideas]
I found this interesting [reddit post](https://www.reddit.com/r/neovim/comments/zk4bjk/quick_tip_one_keymap_can_save_you_from_using/) that leverages hop.nvim for similar functionality of tabout, vim-matchup and nvim-treesitter-textobjects. 
One comment down below also shows using a builtin function `searchpair()` can do similar things.
TBH, I didn't use tabout, vim-matchup and nvim-treesitter-textobjects' function that much, so I can't be sure about the coverage of this "quick tip" method.
@ayamir @Jint-lzxy How do you guys thinks? Maybe we can streamline some of the plugins?

**Comment by ayamir** [comment]
Tried it. But I think it's not natural and intuitive to use `hop` to support it. Because matchup and textobjects are based on `treesitter` which means corresponding jump logic is based on grammer. Compared to simple pattern search, grammar-based filtering is nessceary.

**Reply by charliie-dev** [reply]
Reasonable, got it!
